tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -0,0 +1,163 @@
+// <copyright file="HotChocolateCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate
+{
+    internal class HotChocolateCommon : GraphQLCommonBase
+    {
+        internal const string IntegrationName = nameof(Configuration.IntegrationId.HotChocolate);
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.HotChocolate;
+
+        private const string ServiceName = "hotchocolate";
```

Comment:
TLDR: I think we should have the same service name and a tag to differentiate both implems. (this is what we do for http basically). WDYT?

My thinking: 
Good question. For dbs, we have the dbtype in the service name. But, I assume it's because it's really a different underlying system and here it would make more sense to have graphql? As a user, I wouldn't see the value of differentiating graphql and hotchocolate in that case, so I think it makes sense indeed. That said, if we don't have the service name, what actual data can the user see on his end to differentiate graphql vs hotchocolate use cases? Actually, we can do the same as http, where we have the same service name and a specific tag per implementation.

Reply:
> if we don't have the service name, what actual data can the user see on his end to differentiate graphql vs hotchocolate use cases

We already attach a `component` tag to the span which differentiates between hotchocolate and graphqldotnet. 

> As a user, I wouldn't see the value of differentiating graphql and hotchocolate in that case

Same, graphql clearly describes the _type_ of service, hot chocolate does not.

> Actually, we can do the same as http, where we have the same service name and a specific tag per implementation

That's already what we do, they all get the `http-client` suffix 😉 

Also, these services names are likely all going to change _anyway_, so maybe we should look into what they will be once that RFC is picked up again? 🤔

Reply:
First I'd like to make sure that you understood that I agreed with you @andrewlock :-p. Then:

> That's already what we do, they all get the http-client suffix

Yes we agree. There are multiple http-client (curl, httpclient...) and they all have the same service name xxx-http-client. So as you said, we shall have the same service name.
 
> We already attach a component tag

🤔 , I haven't seen it. That's why I was suggesting to add a specific tag for that as we do in HttpTags.

Reply:
>  I haven't seen it.

We certainly _send_ it, e.g.: 

https://github.com/DataDog/dd-trace-dotnet/blob/97a95eab7133053f6e88b2f27330897a24396cff/tracer/test/snapshots/GraphQL4Tests.SubmitsTraces.verified.txt#L35-L41

so I wonder if it's one of the "hidden" tags?

Reply:
Wow that's cheating, it's in the partial part of the tags... that why I didn't see it :)
Thanks a lot. So indeed no need for the extra tag (which makes me wonder if we really need `http-client-handler-type` tag in http but 🤷‍♂️